depdive/src/github.rs

```diff
@@ -129,6 +132,19 @@ impl GitHubAnalyzer {
         })
     }
 
+    fn make_github_rest_api_call(&self, api_endpoint: &str) -> Result<Response> {
+        let mut response = self.client.get(api_endpoint).send()?;
+        while response.status() == StatusCode::from_u16(429)?
+            || response.status() == StatusCode::from_u16(403)?
+        {
+            // If api rate limit exceeded, sleep for a minute
+            println!("GitHub API rate limit exceeded. Sleeping for a minute");
```

Comment:
how often do we hit the rate limit?

Reply:
In the case of getting commit data in the last six months, we make repeated calls where there's a possibility to hit rate limit. We didn't hit the rate limit while running on `diem`, but hit in some cases while running on `polkadot`. The better solution here is really to use `graphql` as currently we fetch lot of unnecessary data wasting API call counts.